src/PhpParser/AstResolver.php

```diff
@@ -58,7 +60,8 @@ public function __construct(
         private readonly ReflectionProvider $reflectionProvider,
         private readonly NodeTypeResolver $nodeTypeResolver,
         private readonly ClassLikeAstResolver $classLikeAstResolver,
-        private readonly MethodReflectionResolver $methodReflectionResolver
+        private readonly MethodReflectionResolver $methodReflectionResolver,
+        private readonly CurrentFileProvider $currentFileProvider
```

Comment:
This service should be used as little as possible. It juggles a value object, changes state and its not a good practice

Ideally we should remove it.

Reply:
I think we can see this as compromise, we need what file currently rector check.

AstResolver is reading file and decorate from the beginning, while on filename equal, it doesn't make sense, just use what current File object passed, on benefit, it also get latest `->getNewSmts()` which more reliable for current file in case of `--dry-run` usage while cached.

Reply:
I see. In that case we should not make another lock on the file provider, but add caching to `parseFileNameToDecoratedNodes()` method. That way code is fast enough and we have this service decopuled :+1:

Reply:
That already cached, just on very first enter, if there is File object, just use existing

Reply:
I see. In that case we should avoid this, as the `CurrentFileProvider` will be removed to avoid file value juggling.
I'll see where the `AstResolver` is used and how to decreate it.